proposals/new_policy_types.md

```diff
@@ -0,0 +1,956 @@
+    - name: 'check-uid'
+      expression: "request.userInfo.uid == '014fbff9a07c'"
+    - name: 'check-user-groups'
+      expression: "request.userInfo.groups == ['system:authenticated', 'my-admin-group']"
+    - name: 'check-extra'
+      expression: "request.userInfo.extra == {'some-key': ['some-value1', 'some-value2']}"
+```
+
+Here is an example of a Kyverno rule that matches all Services being created by the user `dave` or `mariam`.
+
+```yaml=
+match:
+  any:
+  - resources:
+      kinds:
+      - Service
+      operations:
+      - CREATE
+    subjects:
+    - kind: User
+      name: dave
+    - kind: User
+      name: mariam
+```
+
+The equivalent VAP match resource would look like this:
+
+```yaml=
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        operations:  ["CREATE"]
+        resources:   ["services"]
+  matchConditions:
+    - name: 'check-username'
+      expression: "request.userInfo.username == 'dave' || request.userInfo.username == 'mariam'"
+```
+
+Here is an example of a Kyverno rule that matches all Pods being created by users whose names start with `not-`.
+
+```yaml=
+match:
+  any:
+  - resources:
+      kinds:
+      - Pod
+    subjects:
+    - kind: User
+      name: not-?*
+```
+
+The equivalent VAP match resource would look like this:
+
+```yaml=
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        operations:  ["CREATE"]
+        resources:   ["pods"]
+  matchConditions:
+    - name: 'check-username'
+      expression: "request.userInfo.username.startsWith('not-')"
+```
+
+Here is an example of a Kyverno rule that matches all Pods being created by `test-sa` ServiceAccount in the `test-ns` Namespace.
+
+```yaml=
+match:
+  any:
+  - resources:
+      kinds:
+      - Pod
+  - subjects:
+    - kind: ServiceAccount
+      name: test-sa
+      namespace: test-ns
+```
+
+The equivalent VAP match resource would look like this:
+
+```yaml=
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        operations:  ["CREATE"]
+        resources:   ["pods"]
+  matchConditions:
+    - name: 'check-username'
+      expression: "request.userInfo.username == 'system:serviceaccount:test-ns:test-sa'"
+```
+
+Here is an example of a Kyverno rule that matches all Pods being created by users in the `dev` group.
+
+```yaml=
+  - match:
+      all:
+      - resources:
+          kinds:
+          - Pod
+        subjects:
+        - kind: Group
+          name: dev
+```
+
+The equivalent VAP match resource would look like this:
+
+```yaml=
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        operations:  ["CREATE"]
+        resources:   ["pods"]
+  matchConditions:
+    - name: 'check-user-groups'
+      expression: "request.userInfo.groups == ['dev']"
+```
+
+### Webhook Configuration
```

Comment:
I added details for webhook configuration via https://github.com/MariamFahmy98/KDP/pull/1.